clang/lib/APINotes/APINotesReader.cpp

```diff
@@ -536,6 +536,28 @@ namespace {
             static_cast<EnumExtensibilityKind>((payload & 0x3) - 1);
       }
 
+      unsigned ImportAsLength =
+          endian::readNext<uint16_t, little, unaligned>(data);
+      if (ImportAsLength > 0) {
+        info.SwiftImportAs = (std::string(reinterpret_cast<const char *>(data),
+                                          ImportAsLength - 1));
+        data += ImportAsLength - 1;
+      }
```

Comment:
What do you think of this pattern?

```suggestion
      if (auto BiasedLength = ImportAsLength - 1) {
        info.SwiftImportAs = (std::string(reinterpret_cast<const char *>(data),
                                          BiasedLength));
        data += BiasedLength;
      }
```

Reply:
This underflows `unsigned` if `ImportAsLength == 0`, we could cast it to signed but I think that would slightly hurt readability.